packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx

```diff
@@ -215,30 +216,62 @@ export const useOrganizationSwitcherContext = () => {
     throw new Error('Clerk: useUserButtonContext called outside OrganizationSwitcher.');
   }
 
-  const navigateAfterSwitchOrganizationUrl = () => navigate(ctx.afterSwitchOrganizationUrl);
-  const navigateAfterCreateOrganizationUrl = () => navigate(ctx.afterCreateOrganizationUrl);
+  const afterCreateOrganizationUrl = ctx.afterCreateOrganizationUrl || displayConfig.afterCreateOrganizationUrl;
+  const afterLeaveOrganizationUrl = ctx.afterLeaveOrganizationUrl || displayConfig.afterLeaveOrganizationUrl;
+
+  const navigateCreateOrganization = () => navigate(ctx.createOrganizationUrl || displayConfig.createOrganizationUrl);
+  const navigateOrganizationProfile = () =>
+    navigate(ctx.organizationProfileUrl || displayConfig.organizationProfileUrl);
+  const navigateAfterSwitchOrganization = () =>
+    ctx.afterSwitchOrganizationUrl ? navigate(ctx.afterSwitchOrganizationUrl) : Promise.resolve();
```

Comment:
:question: Where does the `afterSwitchOrganizationUrl` get set on the `OrganizationSwitcherCtx`?

Reply:
Any props passed to a `clerk-react` component end up in the corresponding context. The `clerk-js` context  gets the props [here](https://github.com/clerkinc/javascript/blob/b0d5e73ff718c43382cbf6f2ad23ae9627a8eb9a/packages/clerk-js/src/ui/portal/index.tsx#L21-L20), however this is only part of a bigger flow. Happy to discuss details

Reply:
Thanks! 

Do we still want the `afterSwitchOrganizationUrl` to be passed as a prop? I thought we were ditching this property altogether. Maybe I understood wrong :thinking:. Please disregard this comment if it's invalid.

Reply:
You're right! We ditched the dashboard path and instead, we don't perform navigation by default. However, we'd still like to  allow the user to explicitly overwrite it using a prop.